spec/amber/cli/commands/generator_spec.cr

```diff
@@ -17,13 +17,122 @@ module Amber::CLI
 
           Amber::CLI::Spec.cleanup
         end
+
+        it "follows naming conventions for all files and class names" do
+          ENV["AMBER_ENV"] = "test"
+          Amber::CLI::MainCommand.run ["new", TESTING_APP]
+          Dir.cd(TESTING_APP)
+
+          camel_case = "PostComment"
+          snake_case = "post_comment"
+          incorrect_case = "Post_comment"
+          display = "Post Comment"
+          class_definition_prefix = "class #{camel_case}"
+          spec_definition_prefix = "describe #{camel_case}"
+
+          [camel_case, snake_case].each do |arg|
+            MainCommand.run ["generate", "scaffold", arg, "name:string"]
+
+            File.exists?("./spec/models/#{snake_case}_spec.cr").should be_true
+            File.exists?("./src/models/#{snake_case}.cr").should be_true
+            File.exists?("./spec/controllers/#{snake_case}_controller_spec.cr").should be_true
+            File.exists?("./src/controllers/#{snake_case}_controller.cr").should be_true
+            File.exists?("./src/views/#{snake_case}/_form.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/edit.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/index.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/new.slang").should be_true
+            File.exists?("./src/views/#{snake_case}/show.slang").should be_true
+
+            File.read("./spec/models/#{snake_case}_spec.cr").should contain spec_definition_prefix
+            File.read("./src/models/#{snake_case}.cr").should contain class_definition_prefix
+            File.read("./spec/controllers/#{snake_case}_controller_spec.cr").should contain spec_definition_prefix
+            File.read("./src/controllers/#{snake_case}_controller.cr").should contain class_definition_prefix
+            File.read("./src/views/#{snake_case}/_form.slang").should contain snake_case
+            File.read("./src/views/#{snake_case}/edit.slang").should contain display
+            File.read("./src/views/#{snake_case}/index.slang").should contain display
+            File.read("./src/views/#{snake_case}/new.slang").should contain display
+            File.read("./src/views/#{snake_case}/show.slang").should contain snake_case
+
+            File.read("./config/routes.cr").should contain "#{camel_case}Controller"
+            File.read("./config/routes.cr").should_not contain "#{incorrect_case}Controller"
+
+            File.delete("./spec/models/#{snake_case}_spec.cr")
```

Comment:
Why are we deleting all of these one by one? We already delete he folder after the test. At least we should be.

Reply:
Looks like you sort of answered my question below. Even if you do have a reason for deleting these we shouldn't be deleting them one by one.

Reply:
I can clean it up, the two (not mutually exclusive) ideas that come to mind first are:
- Use `rm -rf ./spec/*` `rm -rf ./src/*`
- Refactor to put the file handling in a module

Reply:
The first option seems clean and easy to read.

Reply:
For the scaffold, I changed it so that a new app is created for each iteration of the loop, and the spec `cleanup` is also run. This is because the routes file is touched and that can't be deleted safely without generating a new app.

When applicable, I switched to `rm -rf ./[dir]/*`